dotcom-rendering/src/client/ga/index.ts

```diff
@@ -1,44 +1,62 @@
 import {
 	getConsentFor,
+	onConsent,
 	onConsentChange,
 } from '@guardian/consent-management-platform';
 import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
 import { loadScript, log } from '@guardian/libs';
 import { init, sendPageView } from './ga';
 
-export const ga = (): Promise<void> => {
+/** Memoize loading to prevent loading twice */
+let loadedGoogleAnalytics = false;
+
+/** Enable Google Analytics */
+const loadGoogleAnalytics = async () => {
+	if (loadedGoogleAnalytics) return;
+
+	try {
+		await loadScript('https://www.google-analytics.com/analytics.js');
+		loadedGoogleAnalytics = true;
+
+		log('dotcom', 'GA script loaded');
+
+		init();
+		sendPageView();
+	} catch (error) {
+		loadedGoogleAnalytics = false;
+		// We don't need to log script loading errors (these will mostly be adblock, etc),
+		if (!String(error).includes('Error loading script')) {
+			// This is primarily for logging errors with our GA code.
+			window.guardian.modules.sentry.reportError(
+				error instanceof Error ? error : new Error(String(error)),
+				'ga',
+			);
+		}
+	}
+};
+
+/** Disable Google Analytics */
+const unloadGoogleAnalytics = () => {
+	// @ts-expect-error -- We should never be able to directly set things to the global window object
+	// but in this case we want to stub things for testing, so it's ok to ignore this rule
+	window.ga = null;
+	loadedGoogleAnalytics = false;
+};
+
+const handleGoogleAnalytics = async (state: ConsentState) => {
```

Comment:
```suggestion
const manageGoogleAnalytics = async (state: ConsentState) => {
```
'handle' could be a lot of things, but this is deciding whether or not to make GA available. just a suggestion though, not a blocker